### PR TITLE
Issue 8: Moved all functions involved with user input out of main.c and into their own files.

### LINKED ISSUE #8: As a part of issue #1, Take all functions in main.c that handle user input and put them in their own file so that they are easier to view and document.

### PR DESCRIPTION
**Pull request by lukemoore495**
main.c is far too large to easily be read. The purpose of this was to make the functionality better decomposed and documentation more easily understood. Moving things around was a slightly larger endeavor than I expected.

### CODE REVIEW

File: src/chunk.h [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+Chunk *find_chunk(int p, int q, Model *model);
+
+int chunk_distance(Chunk *chunk, int p, int q);
+
+int has_lights(Chunk *chunk);
+
+void dirty_chunk(Chunk *chunk, Model *model);
+
+
+#endif
```
Comment (fosterj14) [inline src/chunk.h:20]: For good style, add a new line here

File: src/hit.h [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    float vx, float vy, float vz,
+    int *hx, int *hy, int *hz);
+
+int hit_test(
+    int previous, float x, float y, float z, float rx, float ry,
+    int *bx, int *by, int *bz, Model *model);
+
+int hit_test_face(Player *player, int *x, int *y, int *z, int *face, Model *model);
+
+#endif
```
Comment (fosterj14) [inline src/hit.h:21]: For good style, add a new line here

File: src/input.h [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+void add_message(const char *text, Model *model);
+
+void login(Model *model);
+
+void parse_command(const char *buffer, int forward, Model *model);
+
+void on_key(GLFWwindow *window, int key, int scancode, int action, int mods, Model *model);
+
+#endif
```
Comment (fosterj14) [inline src/input.h:35]: For good style, add a new line here

File: src/structs.h [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+    int server_port;
+    int day_length;
+    int time_changed;
+    Block block0;
+    Block block1;
+    Block copy0;
+    Block copy1;
+} Model;
+
+#endif
```
Comment (fosterj14) [inline src/structs.h:144]: For good style, add a new line here

**Review by fosterj14 — COMMENTED** [review]
Added comments to add new lines at the ends of some files

File: src/chunk.h (hunk 1 above)
Comment (fosterj14) [inline src/chunk.h]: For good style, add a new line here

File: src/hit.h (hunk 2 above)
Comment (fosterj14) [inline src/hit.h]: For good style, add a new line here

File: src/structs.h (hunk 4 above)
Comment (fosterj14) [inline src/structs.h]: For good style, add a new line here

File: src/input.h (hunk 3 above)
Comment (fosterj14) [inline src/input.h]: For good style, add a new line here

**Review by fosterj14 — APPROVED** [review]
All files that were missing an end line now have end lines. All comments addressed, so will approve request.

**Comment by TiMcCall** [comment]
Everything looked clear and well documented with no conflicting changes.  Merge approved.

**Pull request merged**